src/OVAL/probes/independent/yamlfilecontent_probe.c

```diff
@@ -265,21 +335,37 @@ static int process_yaml_file(const char *prefix, const char *path, const char *f
 			"path", OVAL_DATATYPE_STRING, path,
 			"filename", OVAL_DATATYPE_STRING, filename,
 			"yamlpath", OVAL_DATATYPE_STRING, yamlpath,
-			/*
-			"windows_view",
-			*/
+			// TODO: Implement "windows_view",
 			NULL
 		);
 		while (oscap_iterator_has_more(values_it)) {
-			SEXP_t *value_sexp = oscap_iterator_next(values_it);
-			probe_item_ent_add(item, "value_of", NULL, value_sexp);
+			SEXP_t *result_ent = probe_ent_creat1("value", NULL, NULL);
+			probe_ent_setdatatype(result_ent, OVAL_DATATYPE_RECORD);
+			struct oscap_htable *record = oscap_iterator_next(values_it);
+			struct oscap_htable_iterator *record_it = oscap_htable_iterator_new(record);
+			while(oscap_htable_iterator_has_more(record_it)) {
+				const struct oscap_htable_item *record_item = oscap_htable_iterator_next(record_it);
+				struct oscap_iterator *item_value_it = oscap_iterator_new(record_item->value);
+				SEXP_t se_tmp_mem;
+				SEXP_t *key = SEXP_string_new_r(&se_tmp_mem, record_item->key, strlen(record_item->key));
+				while(oscap_iterator_has_more(item_value_it)) {
+					SEXP_t *value_sexp = oscap_iterator_next(item_value_it);
+					SEXP_t *field = probe_ent_creat1("field", NULL, value_sexp);
+					probe_item_attr_add(field, "name", key);
+					SEXP_list_add(result_ent, field);
+				}
+				oscap_iterator_free(item_value_it);
+				SEXP_free_r(&se_tmp_mem);
+			}
+			oscap_htable_iterator_free(record_it);
+			SEXP_list_add(item, result_ent);
```

Comment:
Please check if result_ent needs to be freed.

Reply:
Yep, thanks